server/src/engine.rs

```diff
@@ -0,0 +1,87 @@
+//! Implementation of pricing logics based on the ride type.
+
+use crate::grpc::PricingRequest;
+
+/// Get pricing for a given query
+///
+/// # Arguments
+/// * `request` - PricingRequest
+pub fn get_pricing(query: PricingRequest) -> f32 {
+    match query.service_type {
+        0 => get_cargo_pricing(query),
+        1 => get_rideshare_pricing(query),
+        2 => get_charter_pricing(query),
+        _ => 0.0,
```

Comment:
Would probably want to return Result<f32, Error> rather than 0, and implement an error message

Though would be quite funny if someone got a $0.00 bill for a ride somehow

I would be fine if this were made into a Dework ticket for the pricing module and postponed: https://app.dework.xyz/arrow-air/pricing

Reply:
Yea this needs to be a separate ticket where there should be a method that validates grpc requests (as per mentioned on Discord).

Nw, on one can get free ride, the lowest cost is like $2 something to cover the takeoff and landing :)

Reply:
https://app.dework.xyz/arrow-air/services-1?taskId=2c53cf60-3c14-4525-8bfc-6423e1ee043e